src/vivarium/framework/population/manager.py

```diff
@@ -140,16 +139,13 @@ class PopulationManager(Manager):
         },
     }
 
-    @property
```

Comment:
This is the primary goal of this PR. We remove the population property (though the attribute still exists) and we leverage the new `get_private_data` method.